streams/src/main/java/org/apache/kafka/streams/kstream/KStreamBuilder.java

```diff
@@ -235,11 +258,15 @@
                                        final Serde<K> keySerde,
                                        final Serde<V> valSerde,
                                        final String... topics) {
-        final String name = newName(KStreamImpl.SOURCE_NAME);
+        try {
+            final String name = newName(KStreamImpl.SOURCE_NAME);
 
-        addSource(offsetReset, name, timestampExtractor, keySerde == null ? null : keySerde.deserializer(), valSerde == null ? null : valSerde.deserializer(), topics);
+            internalTopologyBuilder.addSource(translateAutoOffsetReset(offsetReset), name, timestampExtractor, keySerde == null ? null : keySerde.deserializer(), valSerde == null ? null : valSerde.deserializer(), topics);
```

Comment:
nit: with this many parameters including ternary statements maybe place the params on a separate line for readability?